### PR TITLE
Fix  DcosCloudformationLauncher delete() - QUALITY-2081

### PR DESCRIPTION
**Pull request by cprovencher**
https://jira.mesosphere.com/browse/QUALITY-2081

### CODE REVIEW

File: dcos_launch/platforms/aws.py [hunk 1]
```diff
@@ -21,14 +21,13 @@
 import time
 
 import pkg_resources
+import retrying
 
 import boto3
-import retrying
+import dcos_launch
```
Comment (margaret) [inline dcos_launch/platforms/aws.py]: the import ordering should be grouped by stdlib module, 3rd party module, local module https://www.python.org/dev/peps/pep-0008/#imports (second bullet point)

**Review by margaret — DISMISSED** [review]
Main change looks good! Just one nitpick on import ordering.

File: dcos_launch/aws.py [hunk 2]
```diff
@@ -85,18 +85,21 @@ def describe(self):
...
+        # 2. to make sure it successfully deletes. If not (for example got interrupted by tagging), we retry deleting
+        status = self.stack.wait_for_complete(
+            transition_states=[
+                'CREATE_COMPLETE',
+                'UPDATE_COMPLETE_CLEANUP_IN_PROGRESS',
+                'UPDATE_IN_PROGRESS',
+                'DELETE_IN_PROGRESS'
+            ],
+            end_states=['DELETE_COMPLETE', 'UPDATE_COMPLETE'])
+        if status == 'UPDATE_COMPLETE':
```
Comment (orsenthil) [inline dcos_launch/aws.py:99]: Can you explain via comment as why this should be the case and how did we determine this?  That is state changed to "UPDATE_COMPLETE" implies deletion was interrupted?

File: dcos_launch/platforms/aws.py [hunk 3]
```diff
@@ -296,20 +295,22 @@ def wait_for_complete(self, transition_states: list, end_states: list):
         log.info('Waiting for stack operation to complete')
 
         @retrying.retry(wait_fixed=60 * 1000,
-                        retry_on_result=lambda res: res is False,
+                        retry_on_result=lambda res: not isinstance(res, str),
```
Comment (orsenthil) [inline dcos_launch/platforms/aws.py:298]: This is a week retry logic. If we are not careful, any accidental return value could cause this to retry.

File: dcos_launch/platforms/aws.py [hunk 4]
```diff
@@ -296,20 +295,22 @@ def wait_for_complete(self, transition_states: list, end_states: list):
...
             if stack_status not in transition_states:
                 for event in self.get_stack_events():
                     log.error('Stack Events: {}'.format(event))
                 raise Exception('StackStatus changed unexpectedly to: {}'.format(stack_status))
             log.info('Continuing to wait...')
-            return False
 
-        wait_loop()
+        status = wait_loop()
+        log.info('wait complete')
```
Comment (orsenthil) [inline dcos_launch/platforms/aws.py:312]: Please provide a better message in the logging info.

Comment (margaret) [reply dcos_launch/platforms/aws.py:312]: What info should be added? Could add `Run 'dcos-launch describe' to get node IP addresses.` There's also another logging statement in a different fn that runs after this that says `Cluster is ready!`.

File: dcos_launch/aws.py (hunk 2 above)
Comment (margaret) [reply dcos_launch/aws.py:99]: The comment for it is a couple of lines above https://github.com/dcos/dcos-launch/pull/129/files/5a1a328f6d41f8352654d63421e153907c53a5de#diff-338b729a3fc00f84d642b1018a052e2eR90

File: dcos_launch/platforms/aws.py (hunk 4 above)
Comment (cprovencher) [reply dcos_launch/platforms/aws.py:312]: I'll just remove it in this PR https://github.com/dcos/dcos-launch/pull/130

**Pull request merged**